openedx/openedx-platform
# Allow customizing heartbeat checks

This moves the settings and defaults for the `heartbeat` Django app together with the rest of the LMS settings. Therefore it allows you to add more heartbeat checks by redefining `HEARTBEAT_CHECKS` in `lms.env.json` 

**JIRA tickets**: None
**Discussions**: None
**Dependencies**: None
**Screenshots**: None
**Sandbox URL**: https://pr17146.sandbox.opencraft.hosting/ (e.g. https://pr17146.sandbox.opencraft.hosting/heartbeat ), together with #17147
**Merge deadline**: None

**Testing instructions**:

1. Run this server
2. Check http://localhost:18000/heartbeat , it should work and provide the default checks (modulestore, sql)
3. Run the extended version: http://localhost:18000/heartbeat?extended, you get 1 extra check (celery)
4. Customize the setting: in the LMS server, add a variable in your `lms.env.json`:
```
    "HEARTBEAT_CHECKS": [
        "openedx.core.djangoapps.heartbeat.default_checks.check_modulestore"
    ],
```
5. Restart server to load the new setting
6. Check the URLs again, the sql one shouldn't be there

**Reviewers**
- edX reviewers

**Author concerns**: None
**Settings**: None
